chore(dep): add stringify-package to project source

Merfoo opened the pull request:
Resolves https://github.com/absolute-version/commit-and-tag-version/issues/49

I considered replacing the `stringify-package` package with the `@npmcli/package-json` package as mentioned in https://github.com/absolute-version/commit-and-tag-version/issues/49#issuecomment-1360602538 but decided that it'd be less intrusive to this project's code to just copy over the code and tests over from the  `stringify-package` package instead (https://github.com/npm/stringify-package).

I also thought that adding `@npmcli/package-json` as a dependency a little overkill for this project's singular use case of updating the `package.json` file's version field.

TimothyJones commented:
Thanks for the PR! 

One request - the package you've included is under the [ISC license](https://github.com/npm/stringify-package/blob/main/LICENSE), which we must include if we are distributing a copy. Can you drop that in a comment in the source file, maybe alongside a link to the original?

Otherwise, I like your solution, especially as it's such a simple package - and, as you point out, we don't need all the features of the official npm one.

If ever they change the format and we need to do something more complicated to write to this field, we'd have to use the official one instead. But given that's an unlikely scenario (which would also have other far-reaching changes) paying that cost now seems unnecessary.

Merfoo commented:
I've added the ISC as a comment to the source code and added a link to the `stringify-package` license as well as requested.

(FYI I'm impressed by the fast response to the PR 😄)

TimothyJones commented:
Thanks! I try to respond as soon as I see it, especially if it's a small PR.

Releasing 11.2.1 now.
